electrum/gui/qt/main_window.py

```diff
@@ -2619,8 +2620,35 @@ def settings_dialog(self):
         if d.need_restart:
             self.show_warning(_('Please restart Electrum to activate the new GUI settings'), title=_('Success'))
 
+    def _show_closing_warnings(self) -> bool:
+        """Show any closing warnings and return True if the user chose to quit anyways."""
+        for callback in self.wallet.closing_warning_callbacks:
+            try:
+                warning: Optional[str] = callback()
+            except Exception:
+                self.logger.exception("Closing warning callback failed: ")
+                continue
+            if warning:
+                warning = _("An ongoing operation prevents Electrum from closing:") + "\n\n" + warning
+                buttons = QMessageBox.StandardButton.Cancel | QMessageBox.StandardButton.Close
+                result = self.show_warning(
+                    msg=warning,
+                    title=_("Don't close Electrum yet!"),
+                    buttons=buttons,
+                    defaultButton=QMessageBox.StandardButton.Cancel,
+                )
+                if result == QMessageBox.StandardButton.Cancel:
+                    break
+        else:
+            # user chose to cancel all warnings or there were no warnings
+            return True
+        return False
+
```

Comment:
I would prefer to show a single dialog outside the for loop, showing the list of warnings, rather than several dialogs.